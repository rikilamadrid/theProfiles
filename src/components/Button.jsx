import React from 'react';
/** @jsxImportSource @emotion/react */
import { css, jsx } from '@emotion/react';

const Button = ({ label, onClick, style }) => {
  const stBackButton = css`
    border-radius: 8px;
    border: solid;
    cursor: pointer;
    display: inline-block;
    padding: 9px 12px 8px 12px;
    text-transform: uppercase;
    margin: 5px;
    outline: 0;

    background-color: #1927f0;
    border-color: #1927f0;
    border-width: 1px;
    font-size: 12px;
    color: white;
    font-weight: bold;

    &:hover {
      filter: brightness(80%) saturate(80%);
    }

    &:active {
      filter: brightness(110%) saturate(110%);
    }
  `;
  const clickHandle = (ev) => {
    ev.stopPropagation();
    onClick();
  };

  return (
    <button css={[stBackButton, style]} onClick={clickHandle} aria-label={label}>
      {label}
    </button>
  );
};

export default Button;
