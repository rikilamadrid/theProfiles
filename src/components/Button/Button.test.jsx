import React from 'react';

import { render } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import '@testing-library/jest-dom/extend-expect';
import rerender from 'react-test-renderer';

import Button from './Button';

describe('Button component render', () => {
  test('correctly without crashing, default', () => {
    render(<Button />);
  });

  test('correctly without crashing with testid', () => {
    const { getByTestId } = render(<Button data-testid="btn"/>);
    expect(getByTestId('btn')).toBeDefined();
  });

  test('correctly without crashing, matching snapshot', () => {
    const component = rerender.create(<Button/>);
    expect(component).toMatchSnapshot();
  });
});

describe('Button component property', () => {
  describe('label', () => {
    test('set to any text value', () => {
      const { getByText } = render(<Button label="save" />);
      expect(getByText('save')).toHaveTextContent(/^save$/);
    });
  });
});

describe('Button component interaction', () => {
  test('click', () => {
    const clickCB = jest.fn();
    const { getByTestId } = render(<Button data-testid="btn" onClick={clickCB} />);
    userEvent.click(getByTestId('btn'));
    expect(clickCB).toHaveBeenCalled();
  });
});
