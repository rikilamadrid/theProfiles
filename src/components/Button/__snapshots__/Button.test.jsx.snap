// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button component render correctly without crashing, matching snapshot 1`] = `
<button
  className="css-jldqlh-Button"
  onClick={[Function]}
/>
`;
